### PR TITLE
[CI] macOS Build to buildkite

### PR DESCRIPTION
**Pull request by simon-mo**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by thomasdesr — APPROVED** [review]

**Comment by simon-mo** [comment]
Note to self: we can't merge this yet because `upload_build_info.sh` won't work in the mac instances. They require write access to the bazel_events log bucket.

**Comment by simon-mo** [comment]
Other than that, this PR is in a good shape, we are running medium A-J and C++ tests.

**Pull request merged**
